tqdm/std.py

```diff
@@ -1329,16 +1355,36 @@ def fp_write(s):
         leave = pos == 0 if self.leave is None else self.leave
 
         with self._lock:
+            # decrement instance pos and remove from internal set
+            self._decr_instances(self)
+
             if leave:
                 # stats for overall rate (no weighted average)
                 self.avg_time = None
-                self.display(pos=0)
-                fp_write('\n')
+                self.display()
+                fp_write('\r')
             else:
                 # clear previous display
-                if self.display(msg='', pos=pos) and not pos:
+                if self.display(msg='', pos=pos):
```

Comment:
I am not 100% sure about this change. I don't understand what is `and not pos` here for, but it affects the fourth example from the PR description (with `leave=False`): it leaves an unexpected newline symbol